A6.md

```diff
@@ -0,0 +1,398 @@
+    4. The 10th through 99th retry attempts each add one to the ">=10" count.
+
+For hedged requests, we record the same stats as above, treating the first hedged request as the initial RPC and subsequent hedged requests as retry attempts.
+
+### Configuration Language
+
+Retry and hedging configuration is set as part of the service config, which is transmitted to the client during DNS resolution. Like other aspects of the service config, retry or hedging policies can be specified per-method, per-service, or per-server name.
+
+Service owners must choose between a retry policy or a hedging policy. Unless the service owner specifies a policy in the configuration, retries and hedging will not be enabled. The retry policy and hedging policy each have their own set of configuration options, detailed below. 
+
+The parameters for throttling retry attempts and hedged RPCs when failures exceed a certain threshold are also set in the service config. Throttling applies across methods and services on a particular server, and thus may only be configured per-server name.
+
+#### Retry Policy
+
+This is an example of a retry policy and its associated configuration. It implements exponential backoff with a maximum of three retry attempts, only retrying RPCs when an `UNAVAILABLE` status code is received.
+
+```
+'retry_policy': {
+  'max_retry_attempts': 3
+  'exponential_backoff': {
+    'initial_backoff_ms': 1000,
+    'max_backoff_ms': 5000,
+    'multiplier': 3
+  }
+  'retryable_status_codes': {UNAVAILABLE}
+}
+```
+
+#### Hedging Policy
+
+The following example of a hedging policy configuration will issue up to three hedged requests for each RPC, spaced out at 500ms intervals, until either: one of the requests receives a valid response, all fail, or the overall call deadline is reached. Analogously to `retryable_status_codes` for the retry policy, `non_fatal_status_codes` determines how hedging behaves when a non-OK response is received.
+
+```
+'hedging_policy': { 
+  'max_requests': 3, 
+  'hedging_delay_ms': 500,
+  'non_fatal_status_codes': {UNAVAILABLE, INTERNAL, ABORTED}
+}
+```
+
+The following example issues three hedged requests simultaneously:
+
+```
+'hedging_policy': { 
+  'max_requests': 3,
+  'hedging_delay_ms': 0,
+  'non_fatal_status_codes': {UNAVAILABLE, INTERNAL, ABORTED}
+}
+```
+
+#### Throttling Configuration
+Throttling configuration applies to all services and methods on a given server, and so can only be set per-server name. The following configuration throttles retry attempts and hedged RPCs when the ratio of failures to successes exceeds ~10%.
+
+```
+'retry_throttling': {
+    'max_tokens': 10,
+    'token_ratio': 0.1
+}
+```
+
+#### Integration with Service Config
+
+The retry policy is transmitted to the client through the service config mechanism. The following is what the JSON configuration file would look like:
+
+```
+{
+  'loadBalancingPolicy': string,
+
+  # Throttling parameters for retry attempts and hedged RPCs.
```

Comment:
Done (aside: https://github.com/grpc/grpc/blob/master/doc/service_config.md uses `#` for the comments)